apps/src/blockly/addons/cdoSerializationHelpers.js

```diff
@@ -176,3 +196,25 @@ function reorderBlocks(blocks, blockOrderMap) {
 
   return orderedBlocks;
 }
+
+export function getCombinedSerialization(
+  mainWorkspaceSerialization,
+  hiddenWorkspaceSerialization
+) {
+  if (_.isEmpty(hiddenWorkspaceSerialization)) {
+    return mainWorkspaceSerialization;
+  }
+
+  const combinedSerialization = _.cloneDeep(mainWorkspaceSerialization);
+  combinedSerialization.blocks.blocks = _.unionBy(
```

Comment:
is it possible for `mainWorkspaceSerialization.blocks` to be `null`?

Reply:
It's possible for it to be undefined, if there are no blocks on the main workspace. This is very unlikely, but not impossible, since we always (?) have an undeletable `when run` block or equivalent on the workspace.

Reply:
Added a check here.